File: Lunchtime/Lib/UIAlertController+Extras.m
//
//  UIAlertController+Extras.m
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-16.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

#import "UIAlertController+Extras.h"

@implementation UIAlertController (Extras)

+ (instancetype)alertControllerWithTitle:(NSString *)title preferredStyle:(UIAlertControllerStyle)preferredStyle {
    return [UIAlertController alertControllerWithTitle:title message:nil preferredStyle:preferredStyle];
}

- (void)addDefaultAction:(NSString *)title withHandler:(void (^)(UIAlertAction *action))handler {
    [self addAction:[UIAlertAction actionWithTitle:title style:UIAlertActionStyleDefault handler:handler]];
}

- (void)addCancelAction {
    [self addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:nil]];
}

@end

File: Lunchtime/Lib/UIAlertController+Extras.swift
//
//  UIAlertController+Extras.swift
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-17.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

import UIKit

public extension UIAlertController {

    // MARK: - Convenience Initialiser
    convenience init(title: String? = nil, preferredStyle: UIAlertControllerStyle) {
        self.init(title: title, message: nil, preferredStyle: preferredStyle)
    }

    static func controller(title: String? = nil, preferredStyle: UIAlertControllerStyle) -> UIAlertController {
        return UIAlertController(title: title, preferredStyle: preferredStyle);
    }

    // MARK: - Convenience Actions
    func addAction(title title: String, style: UIAlertActionStyle, handler: (() -> Void)? = nil) {
        addAction(UIAlertAction(title: title, style: style, handler: { _ in handler?() }))
    }

    func addDestructiveAction(title title: String, handler: (() -> Void)? = nil) {
        addAction(UIAlertAction(title: title, style: .Destructive, handler: { _ in handler?() }))
    }

    func addDefaultAction(title title: String, handler: (() -> Void)? = nil) {
        addAction(UIAlertAction(title: title, style: .Default, handler: { _ in handler?() }))
    }

    func addCancelAction(title: String = "Cancel", handler: (() -> Void)? = nil) {
        addAction(UIAlertAction(title: title, style: .Cancel, handler: { _ in handler?() }))
    }
}